x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/pages/coverage_overview/helpers.test.ts

```diff
@@ -45,16 +45,64 @@ describe('helpers', () => {
       expect(getNumOfCoveredSubtechniques(payload)).toEqual(0);
     });
 
-    it('returns number of subtechniques when present', () => {
+    it('returns total number of unique enabled and disabled subtechniques when no filter is passed', () => {
       const payload = {
         ...getMockCoverageOverviewMitreTechnique(),
         subtechniques: [
           getMockCoverageOverviewMitreSubTechnique(),
-          getMockCoverageOverviewMitreSubTechnique(),
+          { ...getMockCoverageOverviewMitreSubTechnique(), id: 'test-id' },
         ],
       };
       expect(getNumOfCoveredSubtechniques(payload)).toEqual(2);
     });
+
+    it('returns total number of unique enabled and disabled subtechniques when both filters are passed', () => {
+      const payload = {
+        ...getMockCoverageOverviewMitreTechnique(),
+        subtechniques: [
+          getMockCoverageOverviewMitreSubTechnique(),
+          { ...getMockCoverageOverviewMitreSubTechnique(), id: 'test-id' },
```

Comment:
I think these tests need some refactoring later. The functions that create mocks are not parameterized and return implicit data. This means that we set the frontend-side CoverageOverview model in these tests implicitly, while the assertions are explicit. It's hard to read such tests because you always need to jump to the mock functions implementation and compare it with the assertions. Also, it makes the tests setup too rigid and we're probably missing a few tests cases because of that. For example, the mock data makes the same rule both enabled and disabled at the same time, which is 1) incorrect and 2) we need to test more than one rule mapped to the same technique through 2 or more tactics, and also test different combinations of these rules being enabled and disabled.

I'm not suggesting to address this in this PR, but this is something to work on later when we get back to working on the next milestone for the Coverage epic.